tfjs-backend-webgpu/src/shader_preprocessor.ts

```diff
@@ -294,21 +300,23 @@ function getSamplerFromInInfo(inInfo: InputInfo, isVec4: boolean): string {
     `;
   }
 
+  const shapeStr = `${texName.charAt(0).toLowerCase() + texName.slice(1)}Shape`;
+
   if (isVec4) {
     return `
-    vec4 ${funcName}(${inputs}) {
-      return ${texName}[getFlatIndex(${type}(${dims.join(',')}),
-        ${getShapeCoords(inInfo.shape)}) / 4];
-    }
-  `;
+      vec4 ${funcName}(${inputs}) {
+        return ${texName}[getFlatIndex(${type}(${dims.join(',')}),
+          ${shapeStr}) / 4];
+      }
+      `;
   }
 
   return `
     float ${funcName}(${inputs}) {
       return float(${texName}[getFlatIndex(${type}(${dims.join(',')}),
-        ${getShapeCoords(inInfo.shape)})]);
+        ${shapeStr})]);
     }
-  `;
+   `;
 }
 
 function getSamplerAtOutputCoords(
```

Comment:
For `const broadcastDims = backend_util.getBroadcastDims(inInfo.shape, outShape);`, `broadcastDims` is also a constant value and not impacted by inRank or outRank. It should be also part of the shader key.

Reply:
done